problem_builder/public/js/plot.js

```diff
@@ -1,5 +1,23 @@
 function PlotBlock(runtime, element) {
 
+    // jQuery helpers
+
+    jQuery.fn.isEmpty = function() {
+        return !$.trim($(this).html());
+    };
+
+    jQuery.fn.isHidden = function() {
+        // Don't use jQuery :hidden selector here;
+        // this is necessary to ensure that result is independent of parent visibility
+        return $(this).css('display') === 'none';
```

Comment:
You're sure there's no way to use :hidden? This seems a bit redundant. If there isn't, then ok. Same question with isVisible down below.